# Profitable App Profile for the Apple Store and Google Play Markets

The project Profitable App Profile For the Apple Store and Google Play Markets has a goal of finding the apps which can create more revenue for the company. Result set will assist us in finding what type of apps are likely to attract more users.

We only build apps that are free to download and install, and our main source of revenue consists of in-app ads. This means our revenue for any given app is mostly influenced by the number of users who use our app — the more users that see and engage with the ads, the better. Our goal for this project is to analyze data to help our developers understand what type of apps are likely to attract more users. To Do so, We will be analysing the data about number of google apps and apple store apps.

* We have data from Apple Store captured in AppleStore.csv.
* AppleStore.csv has data in form of rows and coloumns.
* We have created list of list using dataset stored in AppleStore.csv. Each individual list corresponds to each row in our csv file.

In [1]:
opened_file = open('C:\\Users\shaveta\Datasets\AppleStore.csv',encoding='utf8')
from csv import reader
read_file = reader(opened_file)
Apple_apps_data = list(read_file)

* Similarly, we have data from Google Play Store in googleplaystore.csv.
* We have created list of list using dataset stored in csv file.

In [2]:
opened_file = open('C:\\Users\shaveta\Datasets\googleplaystore.csv',encoding='utf8')
read_file = reader(opened_file)
Google_apps_data = list(read_file)

## Exploring the DataSet

* To explore the data, we have written a function, which displays header information along with few rows to explore.
* It also displays the Number of headers and coloums in dataset.

In [3]:
def explore_data(dataset,start,end,rows_and_coloums=False):
    new_dataset = dataset[start:end]
    for row in new_dataset:
        print(row)
        print("\n")
    if rows_and_coloums:
        print("Number of rows:",len(dataset))
        print("Number of coloumns:",len(dataset[0]))
        print("\n")

* We have explored the both the datasets by calling the explore function which will give us the brief information
  about the each dataset.

In [4]:
print("Google Store Apps Data:")
print(Google_apps_data[0])
explore_data(Google_apps_data,4,10,True)

print("Apple Store Apps Data:")
print(Apple_apps_data[0])
explore_data(Apple_apps_data,20,25,True)

Google Store Apps Data:
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


['Smoke Effect Photo Maker - Smoke Editor', 'ART_AND_DESIGN', '3.8', '178', '19M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'April 26, 2018', '1.1', '4.0.3 and up']


['Infinite Painter', 'ART_AND_DESIGN', '4.1', '36815', '29M', '1,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'June

## Removing the Erroneous Data from DataSet

* The dataset printed above will help us identify the columns which will help us in identifying the high profitable
  apps.
* In Google store app data we can use Rating, Reviews, Installs and Content rating as our measuring criteria for apps.
* In Case of Apple store app data we have rating count, user rating count to do the job of analysation.
* If we are not able to identify the columns actual description, we can always refer to the below links to understand description of each column.
* Links: 
    1. [https://www.kaggle.com/lava18/google-play-store-apps/home]
    2. [https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home]

* After analysing the data in Google Play Store List, we have observed there are certain rows which does not have data filled in all of their coressponding coloums.

In [5]:
print(Google_apps_data[10473])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


* We don't want erroneous data in our dataset as it could lead to inprecise result.
* Thus, we have deleted row indexed at 10473 from our dataset.

In [6]:
print(len(Google_apps_data))
del Google_apps_data[10473]
print(len(Google_apps_data))

10842
10841


## Removing the Duplicate Rows

* Google apps data set also contains duplicate data for each row.
* We have depicted the duplicate rows using the below code.
* We have looped through Google App Store dataset excluding the Header Row.
* Within each loop we have checked if the app name is "Instagram", If yes, we will print the entire row.

In [7]:
for row in Google_apps_data[1:]:
    if row[0] == "Instagram":
        print(row)
        print("\n")

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']




* As we can see above we have multiple entries for a single APP.
* Now we have counted number of duplicate rows in our dataset, So that we can work on unique set of rows in our dataset.

In [8]:
duplicate=[]                          # To Store the Duplicate Value of each Row.
unique=[]                             # To Store each row uniquely.

for row in Google_apps_data[1:]:
    name = row[0]                     # Store the App Name.
    if name in unique:                # Check if App name exists in unique.
        duplicate.append(name)        # If yes, Its duplicate row.
    else:
        unique.append(name)           # If no, Its unique row.
        
print("Count of Duplicate Rows:",len(duplicate))    # count of duplicate rows.
print("Count of unique Rows:", len(Google_apps_data) - len(duplicate)) # This Count includes Header Field.

Count of Duplicate Rows: 1181
Count of unique Rows: 9660


* We dont want to count duplicate apps entry when we analyse the data, 
  so we need to remove the duplicate entries and keep only one entry per app.
* If we examine the rows printed for the Instagram app, the main difference happens on the fourth position of each row, which corresponds to the number of reviews.
* Higher the number of reviews, the more recent the data should be. 
* We have to make sure that we only include unique rows with highest number of reviews.
* Therefore we will create a new dataset which contains unique rows.

In [9]:
reviews_max={}                                   # Reviews_max will have unique app names as key and thier corresponding reviews
                                                 # as values.
for row in Google_apps_data[1:]:                 # Loop through the google store dataset excluding header
    name = row[0]                                # Store name of app in name
    n_reviews = float(row[3])          
    if name in reviews_max:
        if float(row[3]) > reviews_max[name]:    # If name of the name exists in dictionary, make sure to store max review values
            reviews_max[name] = float(row[3])             
    else:
        reviews_max[name] = float(row[3])               # If name odes not exists then add it to dictionary.
print(len(reviews_max))                                 # This is same number of apps which remained after duplicate apps were extracted.

9659


* Using the above code we have extracted a dictionary which we have used extract the unique apps with highest
number of reviews.
* Next, we have removed the duplicate rows from the orignal dataset and created new dataset which consist of rows with unique app names.

In [10]:
android_clean= []                                  # To store our new cleaned up dataset
already_added = []                                 # Which will store the app names

for row in Google_apps_data[1:]:
    name = row[0]
    n_reviews = float(row[3])
    if n_reviews == reviews_max[name] and name not in already_added:         # Check if the reviews in the row equals the max
        android_clean.append(row)                                            # reviews for the same app stored in the reviwes max dictionary.
        already_added.append(name)                                           # If yes, then store that row in the new cleaned list.
        
print(len(android_clean))
explore_data(android_clean,0,5,True)

9659
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Number of rows: 9659
Number of coloumns: 13




## Removing the Non-English Apps from DataSet

* In our analysation, we only want to consider the data which belongs to the apps having language as "English".
* However, both of our datasets have apps listed in other languages also and since they are not directed towards english speaking audience we have to remove them from dataset.
* Keeping this in mind, we have created a function which will return true/false based on whether app has english as language or not.

In [11]:
def check_language(apps_language):
    for ch in apps_language:
        if ord(ch) > 127:
            return False
    return True

* Since we know, In python each english character's ascii value ranges from 0 to 127. We can use this information to clean our data.
* Function check_language verifies the same, If any character of the passed string has value greater than 127, it returns false else it returns True.
* We need to verify our function by running few basic function call tests.

In [12]:
print("Instagram returns: ",check_language('Instagram'))
print("爱奇艺PPS -《欢乐颂2》电视剧热播 returns: ", check_language('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print("Docs To Go™ Free Office Suite returns: ", check_language("Docs To Go™ Free Office Suite"))
print("Instachat 😜 Returns: ", check_language("Instachat 😜"))

Instagram returns:  True
爱奇艺PPS -《欢乐颂2》电视剧热播 returns:  False
Docs To Go™ Free Office Suite returns:  False
Instachat 😜 Returns:  False


* we saw that the function couldn't correctly identify certain English app names like 'Docs To Go™ Free Office Suite' and 'Instachat 😜'. This is because emojis and characters like ™ fall outside the ASCII range and have corresponding numbers over 127.
* If we're going to use the function we've created, we'll lose useful data since many English apps will be incorrectly labeled as non-English. To minimize the impact of data loss, we'll only remove an app if its name has more than three characters with corresponding numbers falling outside the ASCII range. This means all English apps with up to three emoji or other special characters will still be labeled as English. Our filter function is still not perfect, but it should be fairly effective.



In [13]:
def check_language_modified(apps_language):
    count = 0
    for ch in apps_language:
        if ord(ch) > 127:
            count = count + 1
            
    if count > 3:
        return False
    else:
        return True

* Let's verify our new function with all the test cases.

In [14]:
print("Instagram returns: ",check_language_modified('Instagram'))
print("爱奇艺PPS -《欢乐颂2》电视剧热播 returns: ", check_language_modified('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print("Docs To Go™ Free Office Suite returns: ", check_language_modified("Docs To Go™ Free Office Suite"))
print("Instachat 😜 Returns: ", check_language_modified("Instachat 😜"))

Instagram returns:  True
爱奇艺PPS -《欢乐颂2》电视剧热播 returns:  False
Docs To Go™ Free Office Suite returns:  True
Instachat 😜 Returns:  True


* Since our modified version of function cleared all of our test cases, we can use it to clean our datasets.
* We have looped through both datasets and filtered out non-english speaking apps in another list.
* After that we have explored how many apps are remaining to analyse in both the list.

In [15]:
# Google App store
english_android_apps = []
for row in android_clean:
    if check_language_modified(row[0]):
        english_android_apps.append(row)

# Apple Store Apps
english_apple_apps = []
for row in Apple_apps_data[1:]:
    if check_language_modified(row[2]):
        english_apple_apps.append(row)
        
# Let's Explore
print("Remaining Apps in android dataset:", len(english_android_apps))
print("Remaining Apps in Apple dataset:",len(english_apple_apps))
    

Remaining Apps in android dataset: 9614
Remaining Apps in Apple dataset: 6183


## Isolating the Free Apps

* As our company creates only free apps, we dont want to affect our results with the data of paid apps in dataset.
* Therefore it is important to filter out the data which belongs to paid apps.
* For doing the same, we have looped through both the datasets and then checked thier price status.
* If it is free app, we will append it to our result dataset. 

In [16]:
final_android_dataset = []
final_apple_dataset = []

for row in english_android_apps:
    if row[7] == '0':
        final_android_dataset.append(row)
        
for row in english_apple_apps:
    if row[5] == "0":
        final_apple_dataset.append(row)
        
# Let's Explore
print("Remaining Apps in android dataset:")
explore_data(final_android_dataset,0,5,True)
print("\n")
print("Remaining Apps in Apple dataset:")
explore_data(final_apple_dataset,0,5,True)


Remaining Apps in android dataset:
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Number of rows: 8864
Number of coloumns: 13




Remainin

* We are left with the 3222 and 8864 dataset, which should be fine for our analysis.

## Most Common Apps By Genre

As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:
          1. Build a minimal Android version of the app, and add it to Google Play.
          2. If the app has a good response from users, we develop it further.
          3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.
          
Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by getting a sense of what are the most common genres for each market. For this, we'll need to build frequency tables for the prime_genre column of the App Store data set, and for the Genres and Category columns of the Google Play data set.

We'll build two functions we can use to analyze the frequency tables:

* One function to generate frequency tables that show percentages
* Another function we can use to display the percentages in a descending order

In [17]:
def freq_table(dataset,index):
    table = {}
    
    for row in dataset:
        column = row[index]
        
        if column in table:
            table[column] += 1
        else:
            table[column] = 1
    
    table_percentage = {}
    for key in table:
        percentage = (table[key] / len(dataset) ) * 100
        table_percentage[key] = percentage
    
    return table_percentage

def display_table(dataset,index):
    table_p = freq_table(dataset,index)
    table_p_list = []
    
    for key in table_p:
        tuple_key_value = (table_p[key],key)
        table_p_list.append(tuple_key_value)
    
    table_Sorted = sorted(table_p_list,reverse = True)
    for entry in table_Sorted:
        print(entry[1] ," : " , entry[0])

* We will use these functions to check the frequency table of columns prime_genre and genre, and category.
* Let's start with prime_genre column of the apple store and analyse its results.

In [18]:
display_table(final_apple_dataset,-5)

Games  :  58.16263190564867
Entertainment  :  7.883302296710118
Photo & Video  :  4.9658597144630665
Education  :  3.662321539416512
Social Networking  :  3.2898820608317814
Shopping  :  2.60707635009311
Utilities  :  2.5139664804469275
Sports  :  2.1415270018621975
Music  :  2.0484171322160147
Health & Fitness  :  2.0173805090006205
Productivity  :  1.7380509000620732
Lifestyle  :  1.5828677839851024
News  :  1.3345747982619491
Travel  :  1.2414649286157666
Finance  :  1.1173184357541899
Weather  :  0.8690254500310366
Food & Drink  :  0.8069522036002483
Reference  :  0.5586592178770949
Business  :  0.5276225946617008
Book  :  0.4345127250155183
Navigation  :  0.186219739292365
Medical  :  0.186219739292365
Catalogs  :  0.12414649286157665


##### Analysation of Most Common Apps in Apple Store DataSet:
* AS displayed, genre Games takes up the more than half of the space in the apple store i.e (58%), followed by Entertainment at 7% and photo & Video 4%. Education and social networking apps percentage is almost same in the market i.e.3%. Shopping, Utilities, sports, music takes up almost 2% of the market share.

* The general impression is that most of the apps designed for practical purposes (education, shopping, utilities, productivity, lifestyle) have large share in apple store as compared to the app designed for entertainment such as games, photo and video, social networking, sports, music.

* Apart from genre we also have to take in account the number of users actually using the app. As its not certain if games genre have maximum number of apps on apple store, it would also have maximum number of users downloading them and using them.

Let's continue analysing the data from goole play store processed by genre.

In [19]:
display_table(final_android_dataset,1) #category

FAMILY  :  18.907942238267147
GAME  :  9.724729241877256
TOOLS  :  8.461191335740072
BUSINESS  :  4.591606498194946
LIFESTYLE  :  3.9034296028880866
PRODUCTIVITY  :  3.892148014440433
FINANCE  :  3.7003610108303246
MEDICAL  :  3.531137184115524
SPORTS  :  3.395758122743682
PERSONALIZATION  :  3.3167870036101084
COMMUNICATION  :  3.2378158844765346
HEALTH_AND_FITNESS  :  3.0798736462093865
PHOTOGRAPHY  :  2.944494584837545
NEWS_AND_MAGAZINES  :  2.7978339350180503
SOCIAL  :  2.6624548736462095
TRAVEL_AND_LOCAL  :  2.33528880866426
SHOPPING  :  2.2450361010830324
BOOKS_AND_REFERENCE  :  2.1435018050541514
DATING  :  1.861462093862816
VIDEO_PLAYERS  :  1.7937725631768955
MAPS_AND_NAVIGATION  :  1.3989169675090252
FOOD_AND_DRINK  :  1.2409747292418771
EDUCATION  :  1.1620036101083033
ENTERTAINMENT  :  0.9589350180505415
LIBRARIES_AND_DEMO  :  0.9363718411552346
AUTO_AND_VEHICLES  :  0.9250902527075812
HOUSE_AND_HOME  :  0.8235559566787004
WEATHER  :  0.8009927797833934
EVENTS  :  0.7107400

#### Analysation of Most Common Apps in Google Store DataSet:
* In case of Google store dataset, Family apps constitutes the most amount of precense in app store i.e. 18%. Family apps mostly include apps which are used of the enterrtainment purposes. It is followed by Gaming apps with 9% of the share and then we have tools at 8%, business at 4%. All the productive apps such as productivity, finance, medical and sports stands at 3%.

* We have observed a complete different trend in apple store as compared these results. In Google store we have all the productive apps was head than entertainment apps. Here most of the entertainment apps such as shopping, social travle and local, dating have low percentage as compared to the productive apps.

* But the Family apps genrally constitutes of games of kids, also all the categories are broad in nature. so to confirm the pattern we can see the results of the genre column of the google store dataset.

In [20]:
display_table(final_android_dataset,-4)

Tools  :  8.449909747292418
Entertainment  :  6.069494584837545
Education  :  5.347472924187725
Business  :  4.591606498194946
Productivity  :  3.892148014440433
Lifestyle  :  3.892148014440433
Finance  :  3.7003610108303246
Medical  :  3.531137184115524
Sports  :  3.463447653429603
Personalization  :  3.3167870036101084
Communication  :  3.2378158844765346
Action  :  3.1024368231046933
Health & Fitness  :  3.0798736462093865
Photography  :  2.944494584837545
News & Magazines  :  2.7978339350180503
Social  :  2.6624548736462095
Travel & Local  :  2.3240072202166067
Shopping  :  2.2450361010830324
Books & Reference  :  2.1435018050541514
Simulation  :  2.0419675090252705
Dating  :  1.861462093862816
Arcade  :  1.8501805054151623
Video Players & Editors  :  1.7712093862815883
Casual  :  1.7599277978339352
Maps & Navigation  :  1.3989169675090252
Food & Drink  :  1.2409747292418771
Puzzle  :  1.128158844765343
Racing  :  0.9927797833935018
Role Playing  :  0.9363718411552346
Libraries & D

* As we can observe, the pattern shown in category column is same as gerne column. 
* Only difference between both the columns is gerne has wide range as compared to the category columns and as we are working for the broader perspective we will consider the category column forwarding going.

The frequency tables we analyzed on the previous screen showed us that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and fun apps. Now, we'd like to get an idea about the kind of apps with the most users.

### Finding the most popular Genres

The way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but this information is missing for the App Store data set. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

#### Part One: Apple Store DataSet

* Let's start by generating a frequency table for the prime_genre column to get the unique app genres (below, we'll need to loop over the unique genres). we will use the freq_table() function.

In [21]:
apple_store_genre_freq = freq_table(final_apple_dataset,-5)
print(apple_store_genre_freq)

{'Productivity': 1.7380509000620732, 'Weather': 0.8690254500310366, 'Shopping': 2.60707635009311, 'Reference': 0.5586592178770949, 'Finance': 1.1173184357541899, 'Music': 2.0484171322160147, 'Utilities': 2.5139664804469275, 'Travel': 1.2414649286157666, 'Social Networking': 3.2898820608317814, 'Sports': 2.1415270018621975, 'Health & Fitness': 2.0173805090006205, 'Games': 58.16263190564867, 'Food & Drink': 0.8069522036002483, 'News': 1.3345747982619491, 'Book': 0.4345127250155183, 'Photo & Video': 4.9658597144630665, 'Entertainment': 7.883302296710118, 'Business': 0.5276225946617008, 'Lifestyle': 1.5828677839851024, 'Education': 3.662321539416512, 'Navigation': 0.186219739292365, 'Medical': 0.186219739292365, 'Catalogs': 0.12414649286157665}


In [25]:
avg_user_rating_list = []
for genre in apple_store_genre_freq:
    total = 0
    len_genre = 0
    for row in final_apple_dataset:
        genre_app = row[-5]
        if genre_app == genre:
            user_ratings = float(row[6])
            total = total + user_ratings
            len_genre += 1
    
    avg_user_rating = total / len_genre
    avg_usr_rating_tuple = (avg_user_rating,genre)
    avg_user_rating_list.append(avg_usr_rating_tuple)
    
sorted_list = sorted(avg_user_rating_list,reverse= True)
for row in sorted_list:
    print(row[1]," : ",row[0])

Navigation  :  86090.33333333333
Reference  :  74942.11111111111
Social Networking  :  71548.34905660378
Music  :  57326.530303030304
Weather  :  52279.892857142855
Book  :  39758.5
Food & Drink  :  33333.92307692308
Finance  :  31467.944444444445
Photo & Video  :  28441.54375
Travel  :  28243.8
Shopping  :  26919.690476190477
Health & Fitness  :  23298.015384615384
Sports  :  23008.898550724636
Games  :  22788.6696905016
News  :  21248.023255813954
Productivity  :  21028.410714285714
Utilities  :  18684.456790123455
Lifestyle  :  16485.764705882353
Entertainment  :  14029.830708661417
Business  :  7491.117647058823
Education  :  7003.983050847458
Catalogs  :  4004.0
Medical  :  612.0


* We have apps such as navigation, social Networking, music and weather apps on the top of of list. 
* If we explore navigation apps, we can see maximum number of user reviews are based upon the Google naigation app and Waze. If we exculde these apps its user review ratio will drop significantly.

In [27]:
for app in final_apple_dataset:
    if app[-5] == "Navigation":
        print(app[2]," : ", app[6])

Waze - GPS Navigation, Maps & Real-time Traffic  :  345046
Geocaching®  :  12811
ImmobilienScout24: Real Estate Search in Germany  :  187
Railway Route Search  :  5
CoPilot GPS – Car Navigation & Offline Maps  :  3582
Google Maps - Navigation & Transit  :  154911


* Similar is the case of social networking and music apps. Most of the share of these categories comes from the popular known apps such as facebook, Linkedin, skype, instagram, watsapp, spotify, youtube pandora etc.

In [29]:
for app in final_apple_dataset:
    if app[-5] == "Music":
        print(app[2]," : ", app[6])

Pandora - Music & Radio  :  1126879
Shazam - Discover music, artists, videos & lyrics  :  402925
iHeartRadio – Free Music & Radio Stations  :  293228
Deezer - Listen to your Favorite Music & Playlists  :  4677
Sonos Controller  :  48905
NRJ Radio  :  38
radio.de - Der Radioplayer  :  64
Spotify Music  :  878563
SoundCloud - Music & Audio  :  135744
Sing Karaoke Songs Unlimited with StarMaker  :  26227
SoundHound Song Search & Music Player  :  82602
Ringtones for iPhone & Ringtone Maker  :  25403
Coach Guitar - Lessons & Easy Tabs For Beginners  :  2416
TuneIn Radio - MLB NBA Audiobooks Podcasts Music  :  110420
Magic Piano by Smule  :  131695
QQ音乐HD  :  224
The Singing Machine Mobile Karaoke App  :  130
Bandsintown Concerts  :  30845
PetitLyrics  :  0
edjing Mix:DJ turntable to remix and scratch music  :  13580
Smule Sing!  :  119316
Amazon Music  :  106235
AutoRap by Smule  :  18202
My Mixtapez Music  :  26286
Certified Mixtapes - Hip Hop Albums & Mixtapes  :  9975
Karaoke - Sing Kara

In [28]:
for app in final_apple_dataset:
    if app[-5] == "Social Networking":
        print(app[2]," : ",app[6])

Facebook  :  2974676
LinkedIn  :  71856
Skype for iPhone  :  373519
Tumblr  :  334293
Match™ - #1 Dating App.  :  60659
WhatsApp Messenger  :  287589
TextNow - Unlimited Text + Calls  :  164963
Grindr - Gay and same sex guys chat, meet and date  :  23201
imo video calls and chat  :  18841
Ameba  :  269
Weibo  :  7265
Badoo - Meet New People, Chat, Socialize.  :  34428
Kik  :  260965
Qzone  :  1649
Fake-A-Location Free ™  :  354
Tango - Free Video Call, Voice and Chat  :  75412
MeetMe - Chat and Meet New People  :  97072
SimSimi  :  23530
Viber Messenger – Text & Call  :  164249
Find My Family, Friends & iPhone - Life360 Locator  :  43877
Weibo HD  :  16772
POF - Best Dating App for Conversations  :  52642
GroupMe  :  28260
Lobi  :  36
WeChat  :  34584
ooVoo – Free Video Call, Text and Voice  :  177501
Pinterest  :  1061624
知乎  :  397
Qzone HD  :  458
Skype for iPad  :  60163
LINE  :  11437
QQ  :  9109
LOVOO - Dating Chat  :  1985
QQ HD  :  5058
Messenger  :  351466
eHarmony™ Dating App

Our aim is find popular gerne, but navigation, social networking and music apps might seem more popluar then they really are.The avergae number of rating seems to be skewed by some popular apps which have hundereds of thousands of ratings, while the other apps may struggle to get past 10,000 threshold

Reference apps have avergae user rating of 74942, but its acually bible and dictionary.com which skew up the average ratings.

In [30]:
for app in final_apple_dataset:
    if app[-5] == "Reference":
        print(app[2], " : ",app[6])

Bible  :  985920
Dictionary.com Dictionary & Thesaurus  :  200047
Dictionary.com Dictionary & Thesaurus for iPad  :  54175
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran  :  18418
Merriam-Webster Dictionary  :  16849
Google Translate  :  26786
Night Sky  :  12122
WWDC  :  762
Jishokun-Japanese English Dictionary & Translator  :  0
教えて!goo  :  0
VPN Express  :  14
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition  :  17588
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools  :  4693
Guides for Pokémon GO - Pokemon GO News and Cheats  :  826
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free  :  718
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE)  :  8535
GUNS MODS for Minecraft PC Edition - Mods Tools  :  1497
Real Bike Traffic Rider Virtual Reality Glasses  :  8


However, this niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

* Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app                  adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.
* Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and                drink app requires actual cooking and a delivery service, which is outside the scope of our company.
* Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires                  domain knowledge, and we don't want to hire a finance expert just to build an app.

Now let's analyze the Google Play market a bit.

### Most Popular Apps by Genre on Google Play Store

We have data about the number of installs for the Google Play market, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.)

In [32]:
display_table(final_android_dataset,5) # The installs column

1,000,000+  :  15.726534296028879
100,000+  :  11.552346570397113
10,000,000+  :  10.548285198555957
10,000+  :  10.198555956678701
1,000+  :  8.393501805054152
100+  :  6.915613718411552
5,000,000+  :  6.825361010830325
500,000+  :  5.561823104693141
50,000+  :  4.7721119133574
5,000+  :  4.512635379061372
10+  :  3.5424187725631766
500+  :  3.2490974729241873
50,000,000+  :  2.3014440433213
100,000,000+  :  2.1322202166064983
50+  :  1.917870036101083
5+  :  0.78971119133574
1+  :  0.5076714801444043
500,000,000+  :  0.2707581227436823
1,000,000,000+  :  0.22563176895306858
0+  :  0.04512635379061372
0  :  0.01128158844765343


For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to find out which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on. To perform computations, however, we'll need to convert each install number from string to float. This means we need to remove the commas and the plus characters, otherwise the conversion will fail and raise an error.

Let's start by generating a frequency table for the Category column of the Google Play data set to get the unique app genres (below, we'll need to loop over the unique genres). we can use the freq_table() function

In [34]:
google_app_freq_table = freq_table(final_android_dataset,1) #category
print(google_app_freq_table)

{'ART_AND_DESIGN': 0.6430505415162455, 'AUTO_AND_VEHICLES': 0.9250902527075812, 'BEAUTY': 0.5979241877256317, 'BOOKS_AND_REFERENCE': 2.1435018050541514, 'BUSINESS': 4.591606498194946, 'COMICS': 0.6204873646209386, 'COMMUNICATION': 3.2378158844765346, 'DATING': 1.861462093862816, 'EDUCATION': 1.1620036101083033, 'ENTERTAINMENT': 0.9589350180505415, 'EVENTS': 0.7107400722021661, 'FINANCE': 3.7003610108303246, 'FOOD_AND_DRINK': 1.2409747292418771, 'HEALTH_AND_FITNESS': 3.0798736462093865, 'HOUSE_AND_HOME': 0.8235559566787004, 'LIBRARIES_AND_DEMO': 0.9363718411552346, 'LIFESTYLE': 3.9034296028880866, 'GAME': 9.724729241877256, 'FAMILY': 18.907942238267147, 'MEDICAL': 3.531137184115524, 'SOCIAL': 2.6624548736462095, 'SHOPPING': 2.2450361010830324, 'PHOTOGRAPHY': 2.944494584837545, 'SPORTS': 3.395758122743682, 'TRAVEL_AND_LOCAL': 2.33528880866426, 'TOOLS': 8.461191335740072, 'PERSONALIZATION': 3.3167870036101084, 'PRODUCTIVITY': 3.892148014440433, 'PARENTING': 0.6543321299638989, 'WEATHER': 

In [36]:
avg_user_rating_list = []
for genre in google_app_freq_table:
    total = 0
    len_genre = 0
    for row in final_android_dataset:
        genre_app = row[1]
        if genre_app == genre:
            n_user_rating = row[5]
            n_user_rating = n_user_rating.replace('+','')
            n_user_rating = n_user_rating.replace(',','')
            user_ratings = float(n_user_rating)
            total = total + user_ratings
            len_genre += 1
    
    avg_user_rating = total / len_genre
    avg_usr_rating_tuple = (avg_user_rating,genre)
    avg_user_rating_list.append(avg_usr_rating_tuple)
    
sorted_list = sorted(avg_user_rating_list,reverse= True)
for row in sorted_list:
    print(row[1]," : ",row[0])

COMMUNICATION  :  38456119.167247385
VIDEO_PLAYERS  :  24727872.452830188
SOCIAL  :  23253652.127118643
PHOTOGRAPHY  :  17840110.40229885
PRODUCTIVITY  :  16787331.344927534
GAME  :  15588015.603248259
TRAVEL_AND_LOCAL  :  13984077.710144928
ENTERTAINMENT  :  11640705.88235294
TOOLS  :  10801391.298666667
NEWS_AND_MAGAZINES  :  9549178.467741935
BOOKS_AND_REFERENCE  :  8767811.894736841
SHOPPING  :  7036877.311557789
PERSONALIZATION  :  5201482.6122448975
WEATHER  :  5074486.197183099
HEALTH_AND_FITNESS  :  4188821.9853479853
MAPS_AND_NAVIGATION  :  4056941.7741935486
FAMILY  :  3695641.8198090694
SPORTS  :  3638640.1428571427
ART_AND_DESIGN  :  1986335.0877192982
FOOD_AND_DRINK  :  1924897.7363636363
EDUCATION  :  1833495.145631068
BUSINESS  :  1712290.1474201474
LIFESTYLE  :  1437816.2687861272
FINANCE  :  1387692.475609756
HOUSE_AND_HOME  :  1331540.5616438356
DATING  :  854028.8303030303
COMICS  :  817657.2727272727
AUTO_AND_VEHICLES  :  647317.8170731707
LIBRARIES_AND_DEMO  :  638

As we can see, communication apps are have the most user reviews, but again some popular apps line facebook, instagram, watsapp who have hunderd of thousands of users review, have skewed the overall average of the communication apps.

In [40]:
for app in final_android_dataset:
    if app[1] == "COMMUNICATION":
        print(app[0]," : ", app[5])
        

WhatsApp Messenger  :  1,000,000,000+
Messenger for SMS  :  10,000,000+
My Tele2  :  5,000,000+
imo beta free calls and text  :  100,000,000+
Contacts  :  50,000,000+
Call Free – Free Call  :  5,000,000+
Web Browser & Explorer  :  5,000,000+
Browser 4G  :  10,000,000+
MegaFon Dashboard  :  10,000,000+
ZenUI Dialer & Contacts  :  10,000,000+
Cricket Visual Voicemail  :  10,000,000+
TracFone My Account  :  1,000,000+
Xperia Link™  :  10,000,000+
TouchPal Keyboard - Fun Emoji & Android Keyboard  :  10,000,000+
Skype Lite - Free Video Call & Chat  :  5,000,000+
My magenta  :  1,000,000+
Android Messages  :  100,000,000+
Google Duo - High Quality Video Calls  :  500,000,000+
Seznam.cz  :  1,000,000+
Antillean Gold Telegram (original version)  :  100,000+
AT&T Visual Voicemail  :  10,000,000+
GMX Mail  :  10,000,000+
Omlet Chat  :  10,000,000+
My Vodacom SA  :  5,000,000+
Microsoft Edge  :  5,000,000+
Messenger – Text and Video Chat for Free  :  1,000,000,000+
imo free video calls and chat  

Orfox: Tor Browser for Android  :  10,000,000+
EO Mumbai  :  10+
EP RSS Reader  :  100+
Voxer Walkie Talkie Messenger  :  10,000,000+
ES-1  :  500+
Hangouts Dialer - Call Phones  :  10,000,000+
EU Council  :  1,000+
Council Voting Calculator  :  5,000+
Have your say on Europe  :  500+
Programi podrške EU  :  100+
Inbox.eu  :  10,000+
Web Browser for Android  :  1,000,000+
Everbridge  :  100,000+
Best Auto Call Recorder Free  :  500+
EZ Wifi Notification  :  10,000+
Test Server SMS FA  :  5+
Lite for Facebook Messenger  :  1,000,000+
FC Browser - Focus Privacy Browser  :  1,000+
EHiN-FH conferenceapp  :  100+
Carpooling FH Hagenberg  :  100+
Wi-Fi Auto-connect  :  1,000,000+
Talkie - Wi-Fi Calling, Chats, File Sharing  :  500,000+
WeFi - Free Fast WiFi Connect & Find Wi-Fi Map  :  1,000,000+
Sat-Fi  :  5,000+
Portable Wi-Fi hotspot Free  :  100,000+
TownWiFi | Wi-Fi Everywhere  :  500,000+
Jazz Wi-Fi  :  10,000+
Sat-Fi Voice  :  1,000+
Free Wi-fi HotspoT  :  50,000+
FN Web Radio  :  10+

On an average communication apps have 38,456,119 installs, but this number is skewed by all the apps have user reviews over 1,000,000,000+, 5,000,000+ 10,000,000+ which means if we remove all of these apps, its average review will drop ten times.

In [41]:
app_under_100_million = []

for app in final_android_dataset:
    n_reviews = app[5]
    n_reviews = n_reviews.replace('+','')
    n_reviews = n_reviews.replace(',','')
    reviews = float(n_reviews)
    if app[1] == "COMMUNICATION" and reviews < 100000000:
        app_under_100_million.append(reviews)
        
print(sum(app_under_100_million)/len(app_under_100_million))

3603485.3884615386


We see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

The books and reference genre looks fairly popular as well, with an average number of installs of 8,767,811. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Let's take a look at some of the apps from this genre and their number of installs:

In [43]:
for app in final_android_dataset:
    if app[1] == "BOOKS_AND_REFERENCE":
        print(app[0]," : ", app[5])

E-Book Read - Read Book for free  :  50,000+
Download free book with green book  :  100,000+
Wikipedia  :  10,000,000+
Cool Reader  :  10,000,000+
Free Panda Radio Music  :  100,000+
Book store  :  1,000,000+
FBReader: Favorite Book Reader  :  10,000,000+
English Grammar Complete Handbook  :  500,000+
Free Books - Spirit Fanfiction and Stories  :  1,000,000+
Google Play Books  :  1,000,000,000+
AlReader -any text book reader  :  5,000,000+
Offline English Dictionary  :  100,000+
Offline: English to Tagalog Dictionary  :  500,000+
FamilySearch Tree  :  1,000,000+
Cloud of Books  :  1,000,000+
Recipes of Prophetic Medicine for free  :  500,000+
ReadEra – free ebook reader  :  1,000,000+
Anonymous caller detection  :  10,000+
Ebook Reader  :  5,000,000+
Litnet - E-books  :  100,000+
Read books online  :  5,000,000+
English to Urdu Dictionary  :  500,000+
eBoox: book reader fb2 epub zip  :  1,000,000+
English Persian Dictionary  :  500,000+
Flybook  :  500,000+
All Maths Formulas  :  1,000

The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average:

In [45]:
for app in final_android_dataset:
    if app[1] == "BOOKS_AND_REFERENCE" and (app[5] == '1,000,000,000+' or
                                           app[5] == '500,000,000+' or
                                           app[5] == '100,000,000+'):
        print(app[0]," : ",app[5])

Google Play Books  :  1,000,000,000+
Bible  :  100,000,000+
Amazon Kindle  :  100,000,000+
Wattpad 📖 Free Books  :  100,000,000+
Audiobooks from Audible  :  100,000,000+


* The most popular apps in genre "Books and References" are quite a few therefore as a compitition we dont have to worry about much. 
* Next we can see if rest of the apps, i.e. apps who have number of reviews less than 100,000,000+ are of what kind and what purpose does each one of them serve.
* This will help us stratigies our business plan while developing the app.

In [48]:
for app in final_android_dataset:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

#### Conclusions

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.